Makefile

```diff
@@ -6,15 +6,31 @@ CROSS ?= linux/arm linux/arm64 linux/ppc64le linux/s390x \
 SUDO ?= sudo -n
 
 .PHONY: all
-all: lint test cross
+all: clean lint test test-local cross
+
+.PHONY: clean
+clean:
+	$(RM) mount/go-local.*
 
 .PHONY: test
 test: RUN_VIA_SUDO = $(shell $(SUDO) true && echo -exec \"$(SUDO)\")
-test:
+test: test-local
 	for p in $(PACKAGES); do \
 		(cd $$p && go test $(RUN_VIA_SUDO) -v .); \
 	done
 
+# test the mount module against the local mountinfo source code instead of the
+# release specified in go.mod. This allows catching regressions / breaking
+# changes in mountinfo.
+.PHONY: test-local
+test-local: RUN_VIA_SUDO = $(shell $(SUDO) true && echo -exec \"$(SUDO)\")
+test-local:
+	echo 'replace github.com/moby/sys/mountinfo => ../mountinfo' | cat mount/go.mod - > mount/go-local.mod
+	cd mount \
+	&& go mod download -modfile=go-local.mod \
+	&& go test -modfile=go-local.mod $(RUN_VIA_SUDO) -v .
```

Comment:
I'm not sure if understand this change; `go-local.sum` will be _different_ than `go.sum`, because it's using different versions of the dependency; the `go mod download -modfile=go-local.mod` may pull other versions of dependencies, because `mountinfo` ("main branch") may be using different versions of dependencies.